tox.ini

```diff
@@ -158,8 +158,10 @@ deps =
 extras =
     {[testenv:test]extras}
 passenv = *
+allowlist_externals =
+    {[testenv:test]allowlist_externals}
 commands =
-    {[testenv:test]commands} {posargs} ./test/mockupdb
```

Comment:
posargs is not needed here because its already added via `{[testenv:test]commands}`. Having it here actually duplicates the args, eg see how `--help` becomes `--help --help`:
```
 python -m tox -e test-mockupdb -- --help           
.pkg: _optional_hooks> python /Users/shane/work/pycharm/pymongo-py311/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_sdist> python /Users/shane/work/pycharm/pymongo-py311/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: build_sdist> python /Users/shane/work/pycharm/pymongo-py311/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
test-mockupdb: install_package> python -I -m pip install --force-reinstall --no-deps /Users/shane/git/mongo-python-driver/.tox/.tmp/package/73/pymongo-4.6.0.dev0.tar.gz
test-mockupdb: commands[0]> python --version
Python 3.11.4
test-mockupdb: commands[1]> .evergreen/check-c-extensions.sh
running build_ext
test-mockupdb: commands[2]> pytest -v --durations=5 --maxfail=10 --help --help ./test/mockupdb
usage: pytest [options] [file_or_dir] [file_or_dir] [...]

```